### ISSUE TITLE
Process: Change `Start Process` to return created process object instead of generic handle

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#4104]
Currently if you want to interact with the process started using the `Start Process` keyword you need to do it like this:
```
Start Process    command
${process} =    Get Process Object
# Do something with the process
```
The extra step needed to get the process object is somewhat annoying and it would be much more convenient to just use:
```
${process} =    Start Process    command
```
The problem with this is that `Start Process` currently returns a process handle that can be used with other keywords:
```
${handle} =    Start Process    command
# Other keywords possibly starting more processes
${result} =    Wait For Process    ${handle}
```
The handle is actually just an integer, and it ought to be possible to use the started process object as a handle instead. In practice the above would still work but `${handle}` would be a process object instead of an integer. This would nevertheless be a backwards incompatible change, because someone could have used integers as handles directly:
``` 
Start Process    command
# Other keywords possibly starting more processes
${result} =    Wait For Process    1
```
The above usage hasn't been recommended, though, and aliases should be used instead:
```
Start Process    command    alias=example
# Other keywords possibly starting more processes
${result} =    Wait For Process    example
```

**Comment by pekkaklarck** [comment]
One use case for interacting with started processes is creating pipes between processes. If both #4102 and this issue are implemented, then this would become possible:

```
${process} =    Start Process    command1
${result2} =    Run Process    command2    stdin=${process.stdout}
${result1} =    Wait For Process    ${process}
```

**Comment by pekkaklarck** [comment]
Changing `Start Process` so that it returns the `Popen` object it creates instead of the index of the registered process is easy. The harder part is updating keywords that accept `handle` to work with these returned `Popen` objects. There are two approaches that could be used:

1. Update `ConnectionCache` so that it works not only with indices and aliases but also with objects registered to it. This way keywords possibly wouldn't need to be updated at all.
2. Change every keyword accepting `handle` to check is the `handle` a `Popen` object or not. In the former case they can use the `handle` directly.

**Comment by pekkaklarck** [comment]
I realized implementing this probably isn't too badly backwards incompatible. Even in the case like
```
Start Process    command
# Other keywords possibly starting more processes
${result} =    Wait For Process    1
```
we could still pass that `1` to `ConnectionCache.get_connection()` and it probably would work fine. When used like
```
${process} =    Start Process    command
# Other keywords possibly starting more processes
${result} =    Wait For Process    ${process}
```
the given `handle` is a `Popen` object and we can easily differentiate them from other possible handles.

It could be that only usages where the value returned by `Start Process` itself is expected to be an integer are affected. That shouldn't be a major problem because I see no reason anyone would have used something like this:
```
${handle} =    Start Process    command
Should Be Equal    ${handle}    ${1}
```

**Comment by onurcelep** [comment]
Seems like a new resolver for `Popen` can be implemented and used in `ConnectionCache.get_connection()`
similar to `_resolve_alias` but check if the type of input is `subprocess.Popen`

**Comment by pekkaklarck** [comment]
I don't think there's any reason to make `ConnectionCache` dependent on `subprocess.Popen`. Its code resolving index or alias could just check is the given value an integer or string. If it is neither, then it should check is it one of the registered connections. If it is, then it can be returned as-is. If it's not, then an exception should occur.

**Comment by pekkaklarck** [comment]
This was implemented in great PR #4108 by @onurcelep. Thanks for contribution!
